sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -389,10 +387,7 @@ public ReadWriteSpan recordException(Throwable exception, Attributes additionalA
     StringWriter writer = new StringWriter();
     exception.printStackTrace(new PrintWriter(writer));
     attributes.put(SemanticAttributes.EXCEPTION_STACKTRACE, writer.toString());
-
-    if (additionalAttributes != null) {
```

Comment:
don't we still need to guard against this, since a user can ignore the non-nullability and pass in null?